roles/zuul/files/jobs/dib.yaml

```diff
@@ -0,0 +1,26 @@
+- job:
+    name: 'dib-build'
+    node: 'ubuntu-xenial'
+    builders:
+      shell: |
+        virtualenv $HOME/dib-build
+        source $HOME/dib-build/bin/activate
+        pip install diskimage-builder
+
+        sudo apt-get install -y debootstrap qemu-utils curl \
+                                uuid-runtime parted kpartx
+
+        sudo git clone https://github.com/BonnyCI/hoist.git /opt/hoist
+        export ELEMENTS_PATH=/opt/hoist/roles/nodepool/files/etc/nodepool/elements/nodepool
+        # NOTE: be sure the list of nodepool elements here stays consistent
+        # with the elements specified in the nodepool role.
+        sudo disk-image-create -x -t qcow2 --checksum --no-tmpfs            \
```

Comment:
You loose the virtualenv environment when you sudo here, so disk-image-create is not in $PATH.  Might make more sense to just write the image to /tmp or somewhere in $HOME instead?